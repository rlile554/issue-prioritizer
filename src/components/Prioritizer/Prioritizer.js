import React from 'react';
import { useSelector } from 'react-redux';
import './Prioritizer.scss';
import RepoList from '../RepoList/RepoList';
import IssueList from '../IssueList/IssueList';

const Prioritizer = () => {
	const hasSelectedRepo = useSelector((state) => !!state.selectedRepo);
	return (
		<div className='Content'>
			<RepoList />
			{hasSelectedRepo && <IssueList />}
		</div>
	);
};

export default Prioritizer;
