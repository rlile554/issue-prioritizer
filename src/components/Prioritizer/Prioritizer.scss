.Content {
	display: flex;
	border: 1px solid black;
	border-radius: 5px;
}