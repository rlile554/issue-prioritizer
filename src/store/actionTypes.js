export const ADD_API_KEY = 'ADD_API_KEY';

export const ADD_REPOS = 'ADD_REPOS';
export const SELECT_REPO = 'SELECT_REPO';

export const ADD_ISSUES = 'ADD_ISSUES';
export const REORDER_ISSUES = 'REORDER_ISSUES';

export const LOADING = 'LOADING';