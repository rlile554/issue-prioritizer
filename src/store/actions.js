import {
	ADD_API_KEY,
	ADD_REPOS,
	LOADING,
	SELECT_REPO,
	ADD_ISSUES,
	REORDER_ISSUES,
} from './actionTypes';

export const addApiKey = (apiKey) => {
	return {
		type: ADD_API_KEY,
		apiKey,
	};
};

export const addRepos = (repos) => {
	return {
		type: ADD_REPOS,
		repos,
	};
};

export const selectRepo = (repoId) => {
	return {
		type: SELECT_REPO,
		repoId,
	};
};

export const addIssues = (issues, repoId) => {
	return {
		type: ADD_ISSUES,
		issues,
		repoId,
	};
};

export const reorderIssues = (repoId, sourceIndex, destinationIndex) => {
	return {
		type: REORDER_ISSUES,
		repoId,
		sourceIndex,
		destinationIndex
	}
}

export const loading = (isLoading) => {
	return {
		type: LOADING,
		isLoading,
	};
};
