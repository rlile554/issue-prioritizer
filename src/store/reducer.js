import {
	ADD_API_KEY,
	ADD_REPOS,
	LOADING,
	SELECT_REPO,
	ADD_ISSUES,
	REORDER_ISSUES,
} from './actionTypes';

const initialState = {
	loading: false,
	apiKey: '',
	repos: [],
	issuesList: [],
	selectedRepo: null,
};

const reducer = (state = initialState, action) => {
	switch (action.type) {
		case LOADING:
			return {
				...state,
				loading: action.isLoading,
			};
		case ADD_API_KEY:
			return {
				...state,
				apiKey: action.apiKey,
			};
		case ADD_REPOS:
			return {
				...state,
				repos: action.repos,
				loading: false,
			};
		case SELECT_REPO:
			return {
				...state,
				selectedRepo: action.repoId,
			};
		case ADD_ISSUES:
			const idx = state.issuesList.findIndex((i) => i.id === action.repoId);
			const issuesToAdd = { id: action.repoId, issues: action.issues};
			let newIssues;
			if (idx !== -1) {
				newIssues = [
					...state.issuesList.slice(0, idx),
					issuesToAdd,
					...state.issuesList.slice(idx + 1, state.issuesList.length),
				];
			} else {
				newIssues = [...state.issuesList, issuesToAdd];
			}
			return {
				...state,
				issuesList: newIssues,
			};
		case REORDER_ISSUES:
			const { sourceIndex, destinationIndex, repoId } = action;
			const returnArray = state.issuesList.map((issuesObject) => {
				if (issuesObject.id === repoId) {
					const newIssues = issuesObject.issues.slice();
					const issueToMove = newIssues.splice(sourceIndex, 1)[0];
					newIssues.splice(destinationIndex, 0, issueToMove);
					return {
						...issuesObject,
						issues: newIssues
					};
				}
				return issuesObject;
			});
			return {
				...state,
				issuesList: returnArray,
			};
		default:
			return state;
	}
};

export default reducer;
