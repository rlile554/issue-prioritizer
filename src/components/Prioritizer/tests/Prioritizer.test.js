import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import Prioritizer from '../Prioritizer';

test('Prioritizer renders', () => {
	const component = renderer.create(
		<Provider store={store}>
			<Prioritizer />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
