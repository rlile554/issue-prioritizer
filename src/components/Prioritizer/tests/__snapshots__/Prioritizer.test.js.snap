// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prioritizer renders 1`] = `
<div
  className="Content"
>
  <p>
    Loading
  </p>
</div>
`;
